tensorflow/compiler/mlir/hlo/tests/disc_ral_legalize_to_llvm.mlir

```diff
@@ -0,0 +1,144 @@
+
+    // prepare arguments for the packed ral launch functions
+    // CHECK: %[[KPT0:.*]] = llvm.mlir.constant(12 : i32) : i32
+    // CHECK: %[[KPACK:.*]] = llvm.alloca %[[KPT0]] x !llvm.ptr<i8> : (i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP0:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP0]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP1:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP1]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP2:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP2]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP3:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP3]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP4:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP4]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP5:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP5]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP6:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP6]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP7:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP7]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP8:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP8]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP9:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP9]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP10:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP10]] : !llvm.ptr<ptr<i8>>
+    // CHECK: %[[KP11:.*]] = llvm.getelementptr %[[KPACK]]{{.*}} : (!llvm.ptr<ptr<i8>>, i32) -> !llvm.ptr<ptr<i8>>
+    // CHECK: llvm.store %{{.*}}, %[[KP11]] : !llvm.ptr<ptr<i8>>
+
+    // CHECK: %[[T22:.*]] = llvm.mlir.addressof @ral_kernel_launch___cpu___pvoid_pvoid_pvoid_i64_i64_i64_i64_i64_i64_i32_i32_ppvoid___void : !llvm.ptr<array<90 x i8>>
+    // CHECK: %[[T23:.*]] = llvm.getelementptr %[[T22]]{{.*}} : (!llvm.ptr<array<90 x i8>>, i64, i64) -> !llvm.ptr<i8>
+    // CHECK: llvm.call @disc_ral_call(%[[CTX]], %[[T23]], %[[KPACK]]) : (!llvm.ptr<i8>, !llvm.ptr<i8>, !llvm.ptr<ptr<i8>>) -> ()
+
+    gpu.launch_func  @kernel_module::@the_kernel blocks in (%c1, %c1, %c1) threads in (%c1, %c1, %c1) args(%0 : memref<?x?xf32>)
```

Comment:
Can you split these patterns into different functions and use CHECK-LABEL to match the functions?

CHECK-LABEL delimit sections and makes FileCheck matching a bit more robust and debugging easier.

Reply:
Done!